assets/src/stories-editor/helpers/index.js

```diff
@@ -1718,3 +1718,58 @@ export const processMedia = ( media ) => {
 		poster,
 	};
 };
+
+/**
+ * Copy text to clipboard by using temporary input field.
+ *
+ * @param {string} text Text to copy.
+ */
+export const copyTextToClipBoard = ( text ) => {
+	// Create temporary input element for being able to copy.
+	const tmpInput = document.createElement( 'textarea' );
+	tmpInput.setAttribute( 'readonly', '' );
+	tmpInput.style = {
+		position: 'absolute',
+		left: '-9999px',
+	};
+	tmpInput.value = text;
+	document.body.appendChild( tmpInput );
+	tmpInput.select();
+	document.execCommand( 'copy' );
+	// Remove the temporary element.
+	document.body.removeChild( tmpInput );
+};
+
+/**
+ * Ensure that only allowed blocks are pasted.
+ *
+ * @param {[]}      blocks Array of blocks.
+ * @param {string}  clientId Page ID.
+ * @param {boolean} isFirstPage If is first page.
+ * @return {[]} Filtered blocks.
+ */
+export const ensureAllowedBlocksOnPaste = ( blocks, clientId, isFirstPage ) => {
+	const allowedBlocks = [];
+	// @todo This will need handling for Page Attachment once it's available.
```

Comment:
^ @swissspidy Bringing your attention here just in case -- if/when #3035 gets merged, this would need an update as well to ensure that Page Attachment wouldn't get copied if a CTA block or Attachment already exists on the same Page (no restriction for the first Page).